### PR TITLE
Increase network timeout, small refactors

### PR DESCRIPTION
**Pull request by ZehMatt**
I've noticed that while trying to join a server with a large map and a lot of objects that my game was frozen for quite some time which makes the server think that we disconnected, I've increased the timeout from 7 to 20 seconds which should hopefully cover most cases. I believe some people complained about this before, this time I managed to see it for myself.

### CODE REVIEW

**Review by Gymnasiast — APPROVED** [review]

**Comment by Gymnasiast** [comment]
Would like to see a changelog entry.

File: src/openrct2/network/NetworkConnection.cpp [hunk 1]
```diff
@@ -19,8 +19,9 @@
 
 using namespace OpenRCT2;
 
-constexpr size_t NETWORK_DISCONNECT_REASON_BUFFER_SIZE = 256;
-constexpr size_t NetworkBufferSize = 1024 * 64; // 64 KiB, maximum packet size.
+static constexpr size_t kNetworkDisconnectReasonBufSize = 256;
+static constexpr size_t kNetworkBufferSize = 1024 * 64; // 64 KiB, maximum packet size.
+static constexpr size_t kNetworkNoDataTimeout = 20;     // Seconds.
```
Comment (AaronVanGeffen) [inline src/openrct2/network/NetworkConnection.cpp:24]: Odd, this is used to define `kTimeoutMs` below, but CI still picks it up as unused.

Comment (ZehMatt) [reply src/openrct2/network/NetworkConnection.cpp:24]: Hm yeah..

Comment (ZehMatt) [reply src/openrct2/network/NetworkConnection.cpp:24]: Oh I think its because those are debug builds.

**Review by AaronVanGeffen — APPROVED** [review]

**Pull request merged**